File: implementations/fsharp/src/Skight.AgentPlatform.FSharp/GitTools.fs
namespace Skight.AgentPlatform.FSharp

open System
open System.Text.Json
open System.Threading.Tasks

module GitTools =

    let gitStatus (workspaceRoot: string) : Task<string> =
        TerminalTool.executeCommandDefaultAsync (sprintf "git -C \"%s\" status" workspaceRoot)
        |> Async.StartAsTask

    let gitCommit (workspaceRoot: string) (argsJson: string) : Task<string> =
        async {
            try
                let doc = JsonDocument.Parse argsJson
                use _ = doc
                let root = doc.RootElement
                match root.TryGetProperty("message") with
                | true, msgProp ->
                    let message = msgProp.GetString()
                    let! addResult = TerminalTool.executeCommandDefaultAsync (sprintf "git -C \"%s\" add ." workspaceRoot)
                    if addResult.StartsWith("Error") then
                        return addResult
                    else
                        return! TerminalTool.executeCommandDefaultAsync (sprintf "git -C \"%s\" commit -m \"%s\"" workspaceRoot message)
                | false, _ ->
                    return "Error: Missing 'message' argument for git_commit."
            with ex ->
                return sprintf "Error executing git_commit: %s" ex.Message
        } |> Async.StartAsTask

    let gitPush (workspaceRoot: string) : Task<string> =
        TerminalTool.executeCommandDefaultAsync (sprintf "git -C \"%s\" push" workspaceRoot)
        |> Async.StartAsTask

File: implementations/csharp/src/Skight.AgentPlatform/GitTools.cs
using System;
using System.Text.Json;
using System.Threading.Tasks;

namespace Skight.AgentPlatform
{
    public static class GitTools
    {
        public static async Task<string> GitStatusAsync(string workspaceRoot)
        {
            return await TerminalTool.ExecuteCommandAsync($"git -C \"{workspaceRoot}\" status");
        }

        public static async Task<string> GitCommitAsync(string workspaceRoot, string argsJson)
        {
            try
            {
                using var doc = JsonDocument.Parse(argsJson);
                var root = doc.RootElement;
                if (!root.TryGetProperty("message", out var msgProp))
                {
                    return "Error: Missing 'message' argument for git_commit.";
                }

                var message = msgProp.GetString()!;
                var addResult = await TerminalTool.ExecuteCommandAsync($"git -C \"{workspaceRoot}\" add .");
                if (addResult.StartsWith("Error")) return addResult;

                var commitResult = await TerminalTool.ExecuteCommandAsync($"git -C \"{workspaceRoot}\" commit -m \"{message}\"");
                return commitResult;
            }
            catch (Exception ex)
            {
                return $"Error executing git_commit: {ex.Message}";
            }
        }

        public static async Task<string> GitPushAsync(string workspaceRoot)
        {
            return await TerminalTool.ExecuteCommandAsync($"git -C \"{workspaceRoot}\" push");
        }
    }
}
